### PR TITLE
Issue #9529: Update JavadocMethodCheckTest to use unique Input file

### LINKED ISSUE #9529: update JavadocMethodCheckTest to use unique Input file in each test method
JavadocMethodCheckTest.java file reuse some Input files few time in few different test method.
Expected: each test method in Test class use it own Input file and do not share it with any other test method.

more details - https://github.com/checkstyle/checkstyle/wiki/Good-practice-on-how-to-create-Input-files-for-tests

### PR DESCRIPTION
**Pull request by AkMo3**
Closes #9529

Methods with same input file
1. [testTags()](https://github.com/checkstyle/checkstyle/blob/ebef3bd59fdc2c97587a67bc59d57af6ecbb9072/src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocMethodCheckTest.java#L151) , [testTagsWithResolver()](https://github.com/checkstyle/checkstyle/blob/ebef3bd59fdc2c97587a67bc59d57af6ecbb9072/src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocMethodCheckTest.java#L182), [testTagsWithSubclassesAllowed()](https://github.com/checkstyle/checkstyle/blob/ebef3bd59fdc2c97587a67bc59d57af6ecbb9072/src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocMethodCheckTest.java#L269)
2. [testSetterGetterOff()](https://github.com/checkstyle/checkstyle/blob/ebef3bd59fdc2c97587a67bc59d57af6ecbb9072/src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocMethodCheckTest.java#L357), [testSetterGetterOn()](https://github.com/checkstyle/checkstyle/blob/ebef3bd59fdc2c97587a67bc59d57af6ecbb9072/src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocMethodCheckTest.java#L364)
3. [testGenerics1()](https://github.com/checkstyle/checkstyle/blob/ebef3bd59fdc2c97587a67bc59d57af6ecbb9072/src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocMethodCheckTest.java#L421), [testGenerics2()](https://github.com/checkstyle/checkstyle/blob/ebef3bd59fdc2c97587a67bc59d57af6ecbb9072/src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocMethodCheckTest.java#L434), [testGenerics3()](https://github.com/checkstyle/checkstyle/blob/ebef3bd59fdc2c97587a67bc59d57af6ecbb9072/src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocMethodCheckTest.java#L447)

Duplicate methods have been removed, and only one instance of each of them have been kept.

### CODE REVIEW

**Comment by AkMo3** [comment]
GitHub, rebase

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/InputJavadocMethodExtraThrows.java [hunk 1]
```diff
@@ -12,7 +16,7 @@
      * @throws IllegalArgumentException when argument is wrong // ok
      * @throws NullPointerException indicates null was passed // ok
      */
-    public InputJavadocMethodExtraThrows(String properties) {
+    public InputJavadocMethodExtraThrows(String properties) {                                 // ok
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/InputJavadocMethodExtraThrows.java:19]: I am not sure such alignment at the end is good.
especially when all others are not following this.

please remove extra spaces between code and this traililng comment

**Review by romani — CHANGES_REQUESTED** [review]
for all files:

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/InputJavadocMethodExtraThrows.java (hunk 1 above)
Comment (AkMo3) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/InputJavadocMethodExtraThrows.java]: Such alignment looks a bit disciplined, also moving onto to BDD, visual inspection would be important while reviewing, and in our alignment we might miss some. I have removed right alignment in this PR, but we can follow this approach in next updates.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/InputJavadocMethodIgnoreThrows.java [hunk 2]
```diff
@@ -75,12 +75,12 @@ private static void catchAndRethrowSame(Object o) {
      */
     private static void catchAndRethrowDifferent(Object o, int i) {
         try {
-            float x = 1 / i; // ArithmeticException when i = 0
+            float x = 1 / i;                // ArithmeticException when i = 0
         } catch (RuntimeException ex) {
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/InputJavadocMethodIgnoreThrows.java:79]: Please review all lines, there should be no extra spaces

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/InputJavadocMethodIgnoreThrows.java [hunk 3]
```diff
@@ -126,7 +126,7 @@ private static void ignoreTryWithResources(String input) {
     private static void keepTryWithoutCatch() {
         try (Scanner sc = new Scanner(System.in)) {
             if (sc.nextInt() <= 0) {
-                throw new IllegalArgumentException(""); // violation, not caught and no @param
+                throw new IllegalArgumentException("");    // violation, not caught and no @param
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/InputJavadocMethodIgnoreThrows.java:129]: There extra space.

**Review by romani — CHANGES_REQUESTED** [review]
Please review all files, there still a bunch of not required changes

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/InputJavadocMethodIgnoreThrows.java (hunk 2 above)
Comment (AkMo3) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/InputJavadocMethodIgnoreThrows.java]: Done.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/InputJavadocMethodIgnoreThrows.java (hunk 3 above)
Comment (AkMo3) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/InputJavadocMethodIgnoreThrows.java]: Done.

**Review by romani — APPROVED** [review]

**Pull request merged**